Modules/ScormAicc/scripts/basisAPI.js

```diff
@@ -291,6 +292,26 @@ function status4tree(i_sco,s_status,s_time){
 	}
 }
 
+// Find out if we are in a beforeunload event
+function isBeforeUnloadEvent() {
+	if (window.event && window.event.type === "beforeunload") {
+		return true;
+	}
+	var frame;
+	for (var i = 0; i < window.frames.length; i++) {
+		frame = window.frames[i];
+		if (frame.event && frame.event.type === "beforeunload") {
+			return true;
+		}
+	}
+	return false;
+}
+
+// Find out if we are using Chrome 80+
+function isChromium80plus() {
+	return navigator.userAgent.match(/Chrom(e|ium)\/8\d+/) !== null;
```

Comment:
This regex only matches for Chrome 80-89, is this intended? AFAIK there are no plans to allow these kinds of async requests in the future again and we may have to live with this workaround for quite a while. Depending on the release schedule of Chrome this might break in <1 year.